### PR TITLE
Use the modern tomllib/tomli backends for reading TOML

### PR DESCRIPTION
**Pull request by mgorny**
Replace the deprecated `toml` package with the built-in `tomllib` module in Python 3.11, with fallback to the modern `tomli` package in older Python versions.

### CODE REVIEW

File: pytest_pylint/plugin.py [hunk 1]
```diff
@@ -9,15 +9,20 @@
 from os import getcwd, makedirs, sep
 from os.path import dirname, exists, getmtime, join
 from pathlib import Path
+import sys
```
Comment (michael-k) [inline pytest_pylint/plugin.py:12]: This should go to the top (isort)

File: setup.py [hunk 2]
```diff
@@ -23,7 +23,11 @@
     packages=["pytest_pylint"],
     entry_points={"pytest11": ["pylint = pytest_pylint.plugin"]},
     python_requires=">=3.7",
-    install_requires=["pytest>=5.4", "pylint>=2.3.0", "toml>=0.7.1"],
+    install_requires=[
+        "pytest>=5.4",
+        "pylint>=2.3.0",
+        "tomli; python_version < '3.11'",
```
Comment (michael-k) [inline setup.py:29]:
```suggestion
        "tomli>=1.1.0; python_version < '3.11'",
```

See https://github.com/hukkin/tomli/blob/master/CHANGELOG.md#110

**Review by michael-k — COMMENTED** [review]
2 comments, otherwise LGTM

**Comment by mgorny** [comment]
Thanks for the review. Applied both suggestions.

**Review by michael-k — APPROVED** [review]
LGTM

**Comment by carsongee** [comment]
🤞 this should make this ready to merge. I just dropped a lot of older versions that were failing on your PR. Time to move on!

**Comment by carsongee** [comment]
Success! Thanks for the contribution!

**Comment by mgorny** [comment]
Thank you!

**Pull request merged**